external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -539,3 +569,82 @@ class TosaPartitionPass : public TosaPartitionBase<TosaPartitionPass> {
 std::unique_ptr<Pass> mlir::tosa::createTosaPartitionPass() {
   return std::make_unique<TosaPartitionPass>();
 }
+
+std::unique_ptr<Pass>
+mlir::tosa::createTosaPartitionPass(std::function<bool(Operation *)> &pred,
+                                    std::string attrName, bool noFront) {
+  return std::make_unique<TosaPartitionPass>(pred, attrName, noFront);
+}
+
+
+namespace {
+
+class TestTosaPartitionOptionsPass
```

Comment:
This pass exists only for testing.  Each of its sub-options sets up a particular createTosaPartitionPass() call to test out the argument-passing.  The combinations are arbitrarily arranged to fit the command-line-options test so I can use the same .mlir file for both.